app/routes.py

```diff
@@ -1,2 +1,293 @@
+        return jsonify(sort_goals)
+    
+    else:
+        return jsonify(goals_response)
+
+@goal_bp.route("", methods=["POST"])
+def post_goals():
+        
+    request_body = request.get_json()
+  
+    if "title" not in request_body:
+        return {
+    "details": "Invalid data"
+                },400
+    else:
+        new_goal = Goal(title=request_body["title"])  
+        
+        db.session.add(new_goal) 
+        db.session.commit() 
+        
+        get_goal = Goal.query.get(new_goal.goal_id)
+        
+        return  {
+            "goal": get_goal.goal_to_json()
+        }, 201
+
+
+@goal_bp.route("/<goal_id>", methods=["GET"])
+def get_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    
+    if goal is None:
+        return make_response ("",404)
+    else:
+        return {
+            "goal": goal.goal_to_json()
+        }, 200
+
+@goal_bp.route("/<goal_id>", methods=["PUT"])
+def put_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    
+    if goal is None:
+        return make_response ("",404)
+    else:
+        form_data = request.get_json()
+        
+        goal.title=form_data["title"]
+        
+        
+        #db.session.add(new_data) #adding data to db(a record/row) i.e git add
+        db.session.commit()
+            #new_task = Task.query.get(new_data.task_id)
+        return {
+                "goal": goal.goal_to_json()
+            }, 200
+
+
+@goal_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    
+    if goal is None:
+        return make_response ("",404)
+    else:
+        db.session.delete(goal)
+        
+        db.session.commit()
+     
+        return make_response({
+        "details": f'Goal {goal.goal_id} \"{goal.title}\" successfully deleted'
+        }, 200)
+
+@goal_bp.route("/<goal_id>/tasks", methods=["POST"])
+def post_task_id_to_goal(goal_id):
+    request_body=request.get_json() #information passed in given in json
+    goal=Goal.query.get(goal_id) #grab my goal from db and bring it back
+ 
+    
+    if not goal:
+        return make_response("",404)
+    else:
+        for task_id in request_body["task_ids"]:
+            task=Task.query.get(task_id)
+            task.goal_id=goal_id
+            db.session.commit()
+
+    return {
+        "id": goal.goal_id,
+        "task_ids": request_body["task_ids"]
+    },200
+
+
+@goal_bp.route("/<goal_id>/tasks", methods=["GET"])
+def get_tasks_for_goal(goal_id):
+    
+    goal=Goal.query.get(goal_id) 
+    task = Task.query.get(goal_id)
+    
+    
+    if not goal:
+        return make_response("",404)
+    else:
+
+        outer_dict=goal.goal_to_json()
+        if task==None:
+            outer_dict['tasks']=[]
+        else:
+            inner_dict=task.to_json()
+            
+            inner_dict["goal_id"]=goal.goal_id
+            outer_dict['tasks']=[inner_dict]
+        db.session.commit()
+        return outer_dict,200
```

Comment:
Great job! This was a very clean way of getting the tasks for a specific goal_id